pytorch_lightning/callbacks/pruning.py

```diff
@@ -215,175 +217,183 @@ def _create_pruning_fn(self, pruning_fn: str, *args, **kwargs):
             pruning_fn = _PYTORCH_PRUNING_METHOD[pruning_fn]
             self._global_kwargs = kwargs
             return pruning_fn
-        else:
-            return ModelPruning._wrap_pruning_fn(_PYTORCH_PRUNING_FUNCTIONS[pruning_fn], **kwargs)
+        return ModelPruning._wrap_pruning_fn(_PYTORCH_PRUNING_FUNCTIONS[pruning_fn], **kwargs)
 
     @staticmethod
-    def _wrap_pruning_fn(pruning_fn, *args, **kwargs):
+    def _wrap_pruning_fn(pruning_fn, **kwargs):
         return partial(pruning_fn, **kwargs)
 
-    def _make_pruning_permanent(self):
+    def make_pruning_permanent(self):
+        """ Makes ``parameters_to_prune`` current pruning permanent. """
         for module, param_name in self._parameters_to_prune:
-            pytorch_prune.remove(module, param_name)
-
-    def _resolve_amount(self, current_epoch: int) -> float:
-        if isinstance(self.amount, Callable):
-            amount_fn = self.amount
-            amount = amount_fn(current_epoch)
-        else:
-            amount = self.amount
-        return amount
+            try:
+                pytorch_prune.remove(module, param_name)
+            except ValueError:
+                # pruning already made permanent
+                pass
 
     def _restore_original_weights(self, module: nn.Module, orig_module: nn.Module, tensor_name: str):
-        """
-        "The lottery ticket hypothesis" (https://arxiv.org/pdf/1803.03635.pdf) algorithm:
-
-            1. Randomly initialize a neural network f(x;θ0)(where θ0 ∼Dθ).
-            2. Train the network for j iterations, arriving at parameters θj .
-            3. Prune p% of the parameters in θj , creating a mask m.
-            4. Reset the remaining parameters to their values in θ0, creating the winning ticket f(x; m⊙θ0).
-
-        This function is responsible of step 4.
-        """
         trained = getattr(module, tensor_name)
         orig = getattr(orig_module, tensor_name)
         if trained is None or orig is None:
             return
         trained.data = orig.data.to(trained.device)
 
     def apply_lottery_ticket_hypothesis(self):
-        for (mod, tensor_name), (initial_mod, _) in zip(self._parameters_to_prune, self._initial_parameters_to_prune):
-            self._restore_original_weights(mod, initial_mod, tensor_name)
+        """
+        Lottery ticket hypothesis algorithm (see page 2 of the paper):
+
+            1. Randomly initialize a neural network f(x; θ_0) (where θ_0 ∼ D_θ).
+            2. Train the network for j iterations, arriving at parameters θ_j .
+            3. Prune p% of the parameters in θ_j, creating a mask m.
+            4. Reset the remaining parameters to their values in θ_0, creating the winning ticket f(x; m⊙θ_0).
+
+        This function implements the step 4.
+
+        The ``resample_parameters`` argument can be used to reset the parameters with a new θ_j ∼ D_θ
+        """
+
+        def copy_param(new, old, name: str) -> None:
```

Comment:
Make private.

Reply:
This is a function inside another function. It is private already, can't be accessed from outside.